DEBUGGING.md

```diff
@@ -0,0 +1,625 @@
+# Table of Contents
+1.  [Introduction](#intro)
+2.  [Building Builder](#building-builder)
+3.  [Running Cargo Tests](#cargo-tests)
+4.  [Running Builder](#running-builder)
+5.  [Checking Database](#check-database)
+6.  [Builder UI](#builder-ui)
+7.  [Commonly Encountered Errors](#common-errors)
+8.  [Developer Tips](#dev-tips)
+
+## <a name="intro"></a>
+##Habitat Builder Dev Handbook
+
+This handbook will facilitate a better developer debugging experience with
+the Habitat Builder.  This document will go from start to finish in the
+standalone dev environment to demonstrate some basic functions required to understand and navigate the Builder environment on the front end as well as the back end.  
+
+This document will include general instructions along with a number of repeatable commands or UI actions to ensure your environment is set up correctly.  It will also include troubleshooting information that can be useful in solving common configuration problems.
+
+Also, this document will include a section that can help shorten your
+development time.  Some tips on getting around build errors, what you
+can and can't do if your internet is down, etc. are included.
+
+## <a name="building-builder"></a>
+##Building Builder
+
+Habitat Builder services run inside the hab studio and are mostly Rust services.
+These service are typically built in Linux (18.04) VMs.  The Builder UI runs on the workstation (MacOS) and connects to the Builder services through the builder-api-proxy.  The Builder UI is compiled at launch time and runs in the JavaScript node runtime.  
+   
+This section discusses mainly how to build the Rust based services, but does minimally discuss Builder UI building.  
+
+### Linux Environment
+When building builder, each builder service is built separately.
+Typically, you'll want to clone the repo https://github.com/habitat-sh/builder.git on both the Mac host and on the VM you plan to use for the backend.  This will enable you to run
+Linux builds inside the VM and run the Builder UI from the front-end 
+
+### MacOS Environment
+For the front-ends, kicking off a build is as easy as typing `npm start` from the components/builder-web directory.  Implicit in the 
```

Comment:
It seems like you wanted to add more details but missed it with the statement "Implicit in the"